Sources/SourceKitLSP/DocumentTokens.swift

```diff
@@ -0,0 +1,44 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+
+/// Syntax highlighting tokens for a particular document.
+public struct DocumentTokens {
+  /// Lexical tokens, e.g. keywords, raw identifiers, ...
+  public var lexical: [SyntaxHighlightingToken] = []
+  /// Syntactic tokens, e.g. declarations, etc.
+  public var syntactic: [SyntaxHighlightingToken] = []
+  /// Semantic tokens, e.g. variable references, type references, ...
+  public var semantic: [SyntaxHighlightingToken] = []
+
+  private var merged: [SyntaxHighlightingToken] {
+    [lexical, syntactic, semantic].reduce([]) { $0.mergingTokens(with: $1) }
+  }
+  public var mergedAndSorted: [SyntaxHighlightingToken] {
+    merged.sorted { $0.start < $1.start }
```

Comment:
They aren't (`syntactic` and `semantic` might be, but `lexical` definitely isn't in general, since we update those in deltas by appending to the array).

Perhaps we could perform those updates on a sorted array in a more efficient manner, but I am not sure whether it's worth the complexity there, considering that sorting is also just O(n log n).

Reply:
Yeah, this seems fine for now, definitely simplest.